### PR TITLE
Use the gpu-architecture flag for jit kernels. Fixes TX1 failures

### LINKED ISSUE #1910: JIT Issue on Tegra TX Devices
NVRTC is failing to compile the kernel strings on tegra devices.

```
C++ exception with description "ArrayFire Exception (Internal error:998):
In function std::vector<char> cuda::compileToPTX(const char*, std::__cxx11::string)
In file src/backend/cuda/jit.cpp:264
NVRTC Error(7): NVRTC_ERROR unknown
```
Reported [here](https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/arrayfire-users/vY-Iex7kuE0/EGQ4SxhQBAAJ) initially. We are also able to reproduce this on our side. [Build server log](http://ci.arrayfire.org/job/arrayfire-tegrax1/job/test/job/devel/49/console).

### PR DESCRIPTION
**Pull request by umar456**
This fix addresses the "failed to load builtin" errors on the TX1
hardware. Fixes #1910

### CODE REVIEW

File: src/backend/cuda/jit.cpp [hunk 1]
```diff
@@ -20,6 +20,8 @@
 #include <af/dim4.hpp>
 #include <nvrtc.h>
 
+#include <array>
+#include <cstdio>
```
Comment (pavanky) [inline src/backend/cuda/jit.cpp]: remove pls.

File: src/backend/cuda/jit.cpp [hunk 2]
```diff
@@ -270,8 +273,23 @@ std::vector<char> compileToPTX(const char *ker_name, string jit_ker)
...
     size_t ptx_size;
     std::vector<char> ptx;
-    NVRTC_CHECK(nvrtcCreateProgram(&prog, jit_ker.c_str(), ker_name, 0, NULL, NULL));
-    NVRTC_CHECK(nvrtcCompileProgram(prog, 0, NULL));
+    NVRTC_CHECK(nvrtcCreateProgram(&prog, jit_ker.c_str(),
+                                   ker_name, 0, NULL, NULL));
+
+    auto dev = getDeviceProp(getActiveDeviceId());
+    array<char, 32> arch;
+    snprintf(arch.data(), 32, "--gpu-architecture=compute_%d%d",
```
Comment (pavanky) [inline src/backend/cuda/jit.cpp]: Can you change `32` to `arch.size()` ? May be even change the array size to be a bit bigger. The size of the string is fairly close..

File: src/backend/cuda/jit.cpp (hunk 1 above)
Comment (umar456) [reply src/backend/cuda/jit.cpp]: its for `snprintf`

File: src/backend/cuda/jit.cpp (hunk 2 above)
Comment (umar456) [reply src/backend/cuda/jit.cpp]: Good catch! Fixed

**Comment by umar456** [comment]
build arrayfire ci

**Comment by umar456** [comment]
build arrayfire windows ci
build arrayfire osx cuda ci

**Pull request merged**